flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/split/PulsarPartitionSplitReaderBase.java

```diff
@@ -114,6 +116,9 @@ protected PulsarPartitionSplitReaderBase(
             try {
                 Duration timeout = deadline.timeLeftIfAny();
                 Message<byte[]> message = pollMessage(timeout);
+                if (message == null) {
+                    break;
+                }
 
                 // Deserialize message.
                 collector.setMessage(message);
```

Comment:
why break directly after one message polling failure? should the change here be:
```
if (message != null) {
    collector.setMessage(message)
}
```

This way, the for loop will keep trying until one of the exit condition met

Reply:
The null message indicates that this poll has exceeded the timeout limit, we should switch to another poll.

Reply:
`Duration timeout = deadline.timeLeftIfAny()`  indicates that it met the for loop condition `deadline.hasTimeLeft()`. So we should take another poll.

Reply:
kk, thanks for the clarification